Sources/GRPC/AsyncAwaitSupport/GRPCAsyncRequestStreamWriter.swift

```diff
@@ -90,6 +90,11 @@ public struct GRPCAsyncRequestStreamWriter<Request: Sendable>: Sendable {
     self.asyncWriter.finish()
   }
 
+  /// Sync variant for finishing the request stream for the RPC. This must be called when there are no more requests to be sent.
```

Comment:
Yeah, this makes sense: I'm not surprised here. The compiler will prefer the `async` one in `async` contexts. I think the correct fix is `@disfavoredOverload`, though I don't know if it can override the preference for the `async` method.

Reply:
I suppose that makes sense. I was also wondering about `@disfavoredOverload` but was hoping deprecating the `async` variant may implicitly disfavour it (and there's no reason to use the async version, so deprecating it makes sense).

Reply:
Tried out both approaches with `@_disfavoredOverload` and also `@available(*, deprecated)`. Unfortunately seems both cannot overwrite the preference for `async`. 
Think either we prefix the sync version somehow (like `syncFinish`) so that it can be shipped in a minor version update or we hold this back until the next major version bump.

Reply:
I think our original assertion that removing the `async throws` from finish is API breaking may be incorrect.

`swift package diagnose-api-breaking-changes` does not detect any breaking changes when removing `async throws` (it does if you add them back). The side effect of removing them is warnings for unused occurrence of `try` and `await`.

So -- unless I'm missing something (cc @Lukasa) -- we should just be able to remove `async throws` from the existing `finish()`.

Reply:
I'm open to removing those if we're confident that nothing breaks.

Reply:
Checked that for the internal calls to `finish()` <img width="986" alt="Bildschirmfoto 2022-10-19 um 17 18 29" src="https://user-images.githubusercontent.com/610520/196734597-c6aee45d-1692-45fa-8330-4456b3a29d81.png">